packages/proxy/lib/http/util/cookies.ts

```diff
@@ -0,0 +1,156 @@
+    return 'strict'
+  }
+
+  // being an AUT frame request means it's from navigation, so the context is
+  // 'lax'. otherwise, 'none' indicates a non-navigation cross-origin request
+  return isAUTFrameRequest ? 'lax' : 'none'
+}
+
+const sameSiteNoneRe = /; +samesite=(?:'none'|"none"|none)/i
+
+export const parseCookie = (cookie) => {
+  const toughCookie = Cookie.parse(cookie)
+
+  if (!toughCookie) return
+
+  // fixes tough-cookie defaulting undefined/invalid SameSite to 'none'
+  // https://github.com/salesforce/tough-cookie/issues/191
+  const hasUnspecifiedSameSite = toughCookie.sameSite === 'none' && !sameSiteNoneRe.test(cookie)
+
+  if (hasUnspecifiedSameSite) {
+    toughCookie.sameSite = 'lax'
+  }
+
+  return toughCookie
+}
+
+const comparableCookieString = (toughCookie) => {
+  return _(toughCookie)
+  .pick('key', 'value', 'domain', 'path')
+  .toPairs()
+  .sortBy(([key]) => key)
+  .map(([key, value]) => `${key}=${value}`)
+  .join('; ')
+}
+
+const areCookiesEqual = (cookieA, cookieB) => {
+  return comparableCookieString(cookieA) === comparableCookieString(cookieB)
+}
+
+const matchesPreviousCookie = (previousCookies, cookie) => {
+  return !!previousCookies.find((previousCookie) => {
+    return areCookiesEqual(previousCookie, cookie)
+  })
+}
+
+const toughCookieToAutomationCookie = (toughCookie, defaultDomain) => {
+  const expiry = toughCookie.expiryTime()
+
+  return {
+    domain: toughCookie.domain || defaultDomain,
+    expiry: isFinite(expiry) ? expiry / 1000 : null,
+    httpOnly: toughCookie.httpOnly,
+    maxAge: toughCookie.maxAge,
+    name: toughCookie.key,
+    path: toughCookie.path,
+    sameSite: toughCookie.sameSite,
+    secure: toughCookie.secure,
+    value: toughCookie.value,
+  }
+}
+
+export class CookiesHelper {
+  cookieJar: CookieJar
+  currentAUTUrl: string | undefined
+  request: RequestDetails
+  debug: Debug.Debugger
+  defaultDomain: string
+  sameSiteContext: 'strict' | 'lax' | 'none'
+  previousCookies: Cookie[] = []
+
+  constructor ({ cookieJar, currentAUTUrl, request, debug }) {
+    this.cookieJar = cookieJar
+    this.currentAUTUrl = currentAUTUrl
+    this.request = request
+    this.debug = debug
+    this.sameSiteContext = getSameSiteContext(currentAUTUrl, request.url, request.isAUTFrame)
+
+    const parsedRequestUrl = new URL(request.url)
+
+    this.defaultDomain = parsedRequestUrl.hostname
+  }
+
+  async capturePreviousCookies () {
+    // this plays a part in adding cross-origin cookies to the browser via
+    // automation. if the request doesn't need cross-origin handling, this
+    // is a nooop
+    if (!this.request.needsCrossOriginHandling) return
+
+    this.previousCookies = this.cookieJar.getAllCookies()
+  }
+
+  async getAddedCookies () {
+    // this plays a part in adding cross-origin cookies to the browser via
+    // automation. if the request doesn't need cross-origin handling, this
+    // is a nooop
```

Comment:
```suggestion
    // is a noop
```

Reply:
Addressed in [`52b6874` (#22320)](https://github.com/cypress-io/cypress/pull/22320/commits/52b6874742ee589ed9c40b90792f3074bbf4d426).